Source/Redemption/Private/BTTask_DefineNextStandingPoint.cpp

```diff
@@ -0,0 +1,98 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#include "BTTask_DefineNextStandingPoint.h"
+#include "BehaviorTree/BlackboardComponent.h"
+#include "AIController.h"
+#include "RedemptionPlayer.h"
+#include "Gargoyle.h"
+#include "PatrolPathIndicator.h"
+
+#define GET_PLAYER Cast<ARedemptionPlayer>(OwnerComp.GetBlackboardComponent()->GetValueAsObject(Player.SelectedKeyName))
+#define GET_GARGOYLE Cast<AGargoyle>(OwnerComp.GetAIOwner()->GetPawn())
+
+UBTTask_DefineNextStandingPoint::UBTTask_DefineNextStandingPoint()
+{
+	NodeName = "Define Next Standing Position";
+}
+
+EBTNodeResult::Type UBTTask_DefineNextStandingPoint::ExecuteTask(UBehaviorTreeComponent& OwnerComp, uint8* NodeMemory)
```

Comment:
Would you be able to give just a one line comment before each function to help me understand what they do?

Reply:
execute task is the full execution of the node in the BT, it is what is called when the BT gets to that node.
ClosestPositionToPLayer will find the closest position to the player that the gargoyle can move
SecondClosestPositionToPlayer will find the second closest in case the one that we are currently in is the closest

Reply:
Thanks, would you be able to include that in future code? Just so I know what im looking for when reviewing.

Reply:
ok